### PR TITLE
Delete, Delete in range and Move implemented

### PR DESCRIPTION
**Pull request by AndreAzevedoCoder**
#8 
Added Delete, Delete in Range and Move a point(or a rectangle/circle).
It's pretty simple and maybe not optimized.
Hope it helps someone that just like me need it for a game or something like that.

### CODE REVIEW

**Comment by crhallberg** [comment]
This looks really good! Would it be possible to write a test or two for these changes?

File: quadtree.js [hunk 1]
```diff
@@ -13,11 +13,12 @@ class Point {
 }
 
 class Rectangle {
-  constructor(x, y, w, h) {
+  constructor(x, y, w, h, data) {
     this.x = x;
     this.y = y;
     this.w = w;
     this.h = h;
+    this.userData = data;
```
Comment (crhallberg) [inline quadtree.js]: This change may be useful in the future but it doesn't seem related to the delete and move functionality of this particular pull request. Can we take it out of this branch for now?

Comment (crhallberg) [inline quadtree.js:21]: It looks like below you're using the `userData` of each Point to store an id. Is `userData` on a Rectangle part of it that I didn't see?

File: quadtree.js [hunk 2]
```diff
@@ -276,10 +277,88 @@ class QuadTree {
...
+        deleteArray.push(i)
+      }
+    }
+
+    //When deleted all index change , so the delta fix it
+    var delta = 0
+    for(var i = 0; i < deleteArray.length; i++){
+      this.points.splice(i-delta,1);
+      delta++
+    }
```
Comment (crhallberg) [inline quadtree.js:352]: One approach I like for deleting from an array is to loop backwards thru an array to get around the index changes. Just a tip!

**Comment by crhallberg** [comment]
Hey! I've gotten upgraded permissions to the repository so I'm trying to do some clean up! Any interest in getting this up to date?

**Comment by AndreAzevedoCoder** [comment]
Of course! What can I do?

**Comment by crhallberg** [comment]
Great! I think we left it with two comments that need addressing and missing some tests. Are there any parts of this you want me to tackle?

File: quadtree.js (hunk 1 above)
Comment (crhallberg) [reply quadtree.js:21]: I think this just needs to be removed because it's not being used in the code that I can see.

File: quadtree.js (hunk 2 above)
Comment (crhallberg) [reply quadtree.js:352]: I had a recommendation to simplify this a bit by running backwards thru the array:
```JS
 for( var i = this.points.length; i--;){
    var p = this.points[i];
    if (range.contains(p)) {
      this.points.splice(i, 1);
    }
}
```

With ES6 we can now try to use a filter, which might be cleaner and faster!

**Comment by crhallberg** [comment]
Incorporated into #62, now merged.

**Pull request closed without merging**